README.md
# Banana-Hub

Banana Hub uses Kavo Libary Why? Becuase Im not Wasting my time for a new Roblox UI Libary

How to run Banana-Hub Use Krnl or a Level 7 Thingy
Then you put this script in.

```lua
_G.Key = "Key"

loadstring(game:HttpGet("http://raw.githubusercontent.com/banana-nana/Banana-hub/main/main.lua"))()
```

## Supported Games

Prison Life <br>
Lumber Tycoon

## Planned Things

## Images 



Changelog.md
# Changelog

## Verison 1.1.5

Keys 

## Verison 1

Settings!  Version 1 

## 2022 Jan 14

Changed Things


t.lua
if game.PlaceId == 7256429927 then
    print("Loading Prison Life beta...")
    local Library = loadstring(game:HttpGet("https://raw.githubusercontent.com/xHeptc/Kavo-UI-Library/main/source.lua"))()
    local Window = Library.CreateLib("Prison Life", "Sentinel")
    
    local Main = Window:NewTab("Main")
    local MainS = Main:NewSection("Main")
    MainS:NewDropdown("Give Gun", "Give you a gun", {"M9", "Remington 870", "AK-47"}, function(v)
        local A_1 = game:GetService("Workspace")["Prison_ITEMS"].giver[v].ITEMPICKUP
        local Event = game:GetService("Workspace").Remote.ItemHandler
        Event:InvokeServer(A_1)
    end)
    
    local Player = Window:NewTab("Player")
    local PlayerS = Player:NewSection("Player")

    PlayerS:NewSlider("WalkSpeed", "Changes The Speed", 250, 0, function(v) -- 500 (MaxValue) | 0 (MinValue)
        game.Players.LocalPlayer.Character.Humanoid.WalkSpeed = v
    end)
    PlayerS:NewSlider("JumpPower", "Changes The JumpPower", 250, 0, function(v) -- 500 (MaxValue) | 0 (MinValue)
        game.Players.LocalPlayer.Character.Humanoid.JumpPower = v
    end)


end









Key.lua
_G.Key = "Key1"

local keys = {
    "Banana-nana",
    "FiveDollor",
    "FreeKey"
}

local counter = 1
local keyCheck
for i,v in pairs(keys) do
    if counter == #keys then
        -- Not Vailed Key
        keys = ""
        game.Players.LocalPlayer:Kick("Not a Vailed Key")
    else
        if v == _G.Key then
                
        
        
        if game.PlaceId == 155615604 then
            loadstring(game:HttpGet("http://raw.githubusercontent.com/banana-nana/Banana-hub/main/src/Prisonlife.lua"))()
        end
        
        
        if game.PlaceId == 13822889 then
            loadstring(game:HttpGet("http://raw.githubusercontent.com/banana-nana/Banana-hub/main/src/LumberTycoon.lua"))()
            
        end
        
        
        if game.PlaceId == then
            loadstring(game:HttpGet("http://raw.githubusercontent.com/banan-nana/Banana-hub/main/src/"))()
        end
    
            keyCheck = _G.Key
            keys = ""
        else
            counter = counter +1
        end
    end
end

while true do
    if _G.Key == keyCheck then
        --not spoofed
    else
        game.Player.LocalPlayer:Kick("Bro Had to Spoof")
    end
    wait()
end

src/settings.lua
-- Settings Should be 


game.StarterGui:SetCore("SendNotification", {
    Title = "Banana Hub v1";
    Text = "Yay v1";
    Button1 = "ChangeLog";
    Duration = "4";
    Callback = NotificationBindable;
  })

  function Changelog()
    game.StarterGui:SetCore("SendNotification", {
        Title = "Banana Hub v1";
        Text = "Check Console!";
        Duration = "4";   
    })
  end

  local function callback(Text)
    if Text == "ChangeLog" then
        print("--Changelog--")
        print ("Added Settings -- Note that Only Level 7 can Save Settings")
        print("----------- Updated 2022 Jan 27")
        Changelog()
     end
   end
   
   local NotificationBindable = Instance.new("BindableFunction")
   NotificationBindable.OnInvoke = callback


  local filename = "LumberTycoon_Settings.txt";



  G.SettingTable = {
    WaterSoildL = false;
    setting2 = false;
    setting3 = false;
}


print("Default-Settings")
for i,v in pairs(_G.SettingTable) do 
    print(i,v)
end

function loadsettings()
    print("Loading Settings")
    local HttpService = game:GetService("HttpService")
    if (readfile and isfile and isfile(filename)) then 
        _G.SettingTable = HttpService:JSONDecode(readfile(filename));
        print("settings Loaded")
        print("Settings Are")
        for i,v in pairs(_G.SettingTable) do
            print(i,v)
        end
    end 
end

function saveSettings() 
    print("Saving Settings")
    local json;
    local HttpService = game:GetService("HttpService");
    if (writefile) then
        json = HttpService:JSONEncode(_G.SettingTable);
        writefile(filename, json)
        else
            print(" -- Sorry Cant Save Settings Due to Executor")
    end
end

loadsettings()

src/Prisonlife.lua
local Library = loadstring(game:HttpGet("https://raw.githubusercontent.com/xHeptc/Kavo-UI-Library/main/source.lua"))()
    local Window = Library.CreateLib("Prison Life", "Midnight")
    
    local Main = Window:NewTab("Main")
    local MainS = Main:NewSection("Main")
    MainS:NewDropdown("Give Gun", "Give you a gun", {"M9", "Remington 870", "AK-47"}, function(v)
        local A_1 = game:GetService("Workspace")["Prison_ITEMS"].giver[v].ITEMPICKUP
        local Event = game:GetService("Workspace").Remote.ItemHandler
        Event:InvokeServer(A_1)
    end)
    
    MainS:NewDropdown("Gun Mod", "makes the gun op", {"M9", "Remington 870", "AK-47"}, function(v)
        local module = nil
        if game:GetService("Players").LocalPlayer.Backpack:FindFirstChild(v) then
            module = require(game:GetService("Players").LocalPlayer.Backpack[v].GunStates)
        elseif game:GetService("Players").LocalPlayer.Character:FindFirstChild(v) then
            module = require(game:GetService("Players").LocalPlayer.Character[v].GunStates)   
        end
        if module ~= nil then
            module["MaxAmmo"] = math.huge
            module["CurrentAmmo"] = math.huge
            module["StoredAmmo"] = math.huge
            module["FireRate"] = 0.000001
            module["Spread"] = 0
            module["Range"] = math.huge
            module["Bullets"] = 10
            module["ReloadTime"] = 0.000001
            module["AutoFire"] = true
        end
    end)
    local Player = Window:NewTab("Player")
    local PlayerS = Player:NewSection("Player")

    PlayerS:NewSlider("WalkSpeed", "Changes The Speed", 250, 0, function(v) -- 500 (MaxValue) | 0 (MinValue)
        game.Players.LocalPlayer.Character.Humanoid.WalkSpeed = v
    end)
    PlayerS:NewSlider("JumpPower", "Changes The JumpPower", 250, 0, function(v) -- 500 (MaxValue) | 0 (MinValue)
        game.Players.LocalPlayer.Character.Humanoid.JumpPower = v
    end)

    	local Setting = Window:NewTab("Settings")
      local SettingS = Setting:NewSection("Settings")

      SettingS:NewKeybind("KeyBind", "To Hide", Enum.KeyCode.P, function()
          Library:ToggleUI()
      end)


      G.SettingTable = {
        setting1 = false;
        setting2 = false;
        setting3 = false;
    }
    
    
    print("Default-Settings")
    for i,v in pairs(_G.SettingTable) do 
        print(i,v)
    end
    

src/LumberTycoon.lua
    local Library = loadstring(game:HttpGet("https://raw.githubusercontent.com/xHeptc/Kavo-UI-Library/main/source.lua"))()
    local Window = Library.CreateLib("Lumber Tycoon 2", "Midnight")
    
    local Banana = {
        SchemeColor = Color3.fromRGB(255, 242, 168),
        Background = Color3.fromRGB(255, 242, 168),
        Header = Color3.fromRGB(71, 71, 71),
        TextColor = Color3.fromRGB(71, 71, 71),
        ElementColor = Color3.fromRGB(71, 71, 71)
    }
    
    local Player = Window:NewTab("Player")
    local PlayerS = Player:NewSection("Player")

    PlayerS:NewSlider("WalkSpeed", "Changes The Speed", 2571, 71, function(v) -- 57171 (MaxValue) | 71 (MinValue)
        game.Players.LocalPlayer.Character.Humanoid.WalkSpeed = v
    end)
    PlayerS:NewSlider("JumpPower", "Changes The JumpPower", 2571, 71, function(v) -- 57171 (MaxValue) | 71 (MinValue)
        game.Players.LocalPlayer.Character.Humanoid.JumpPower = v
    end)



    PlayerS:NewToggle("Water Soild", "True or false State ment", function(state)
        if state then
            local WaterSoildL = true
            WaterSoild()
        else
           local WaterSoildL = false
           WaterSoild()
        end
    end)
    PlayerS:NewButton("All Blue Prints", "Gives all bluepritns", function()
        for _,v in pairs(game:GetService("ReplicatedStorage").Purchasables.Structures.BlueprintStructures:GetChildren()) do
            if game:GetService("Players").LocalPlayer.PlayerBlueprints.Blueprints:FindFirstChild(v.Name) == nil then
                local folder = Instance.new("Folder", game:GetService("Players").LocalPlayer.PlayerBlueprints.Blueprints)
                folder.Name = v.Name
                local category = Instance.new("StringValue", folder)
                category.Name = "ItemCategory"
                category.Value = v.ItemCategory.Value
            end
        end
    end)









    local Dupe = Window:NewTab("Dupes")
    local Dupes = Dupe:NewSection("Dupes")



    local Setting = Window:NewTab("Settings")
    local SettingS = Setting:NewSection("Settings")

      SettingS:NewKeybind("KeyBind", "To Hide", Enum.KeyCode.P, function()
        Library:ToggleUI()
      end)  

      SettingS:NewToggle("Notif PlayerAdded", "Sends Notif if Player Join", function(state)
        if state then
            _G.SettingTable.NotificationPlayerAdded = state
        else
            _G.SettingTable.NotificationPlayerAdded = state
        end
        saveSettings()
      end)    







      print("LumberTycoon")




      local filename = "LumberTycoon_Settings.txt";



      _G.SettingTable = {
        WaterSoildL = false;
        NotificationPlayerAdded = true;
        setting3 = false;
    }
    
    
    print("Default-Settings")
    for i,v in pairs(_G.SettingTable) do 
        print(i,v)
    end

    function loadsettings()
        print("Loading Settings")
        local HttpService = game:GetService("HttpService")
        if (readfile and isfile and isfile(filename)) then 
            _G.SettingTable = HttpService:JSONDecode(readfile(filename));
            print("settings Loaded")
            print("Settings Are")
            for i,v in pairs(_G.SettingTable) do
                print(i,v)
            end
        end 
    end

    function saveSettings() 
        print("Saving Settings")
        local json;
        local HttpService = game:GetService("HttpService");
        if (writefile) then
            json = HttpService:JSONEncode(_G.SettingTable);
            writefile(filename, json)
            else
                print(" -- Sorry Cant Save Settings Due to Executor")
        end
    end
    
    loadsettings()

    function WaterSoild() 
        saveSettings()
        if WaterSoildL == true then 
            game:GetService("Workpace").Bridge.VerticalLiftBridge.WaterModel.Water.CanCollide = WaterSoildL
            game:GetService("Workspace").Water.CanCollide = WaterSoildL
        end
        if WaterSoildL == false then 
            game:GetService("Workpace").Bridge.VerticalLiftBridge.WaterModel.Water.CanCollide = WaterSoildL
            game:GetService("Workspace").Water.CanCollide = WaterSoildL
        end
    end


while NotificationPlayerAdded == true do
    wait(2)
    
    game.Players.PlayerAdded:Connect(function(player))
        game.StarterGui:SetCore("SendNotification", {
            Title = "Player Joined - Banana-Hub";
            Text = player.Name .. " Joined";
            Duration = "4";   
        })
    end
end

Main.lua
-- Thank you for using  banana hub
-- Github banana-nana





-- 



if game.PlaceId == 155615604 then
    local Library = loadstring(game:HttpGet("https://raw.githubusercontent.com/xHeptc/Kavo-UI-Library/main/source.lua"))()
    local Window = Library.CreateLib("Prison Life", "Midnight")
    
    local Main = Window:NewTab("Main")
    local MainS = Main:NewSection("Main")
    MainS:NewDropdown("Give Gun", "Give you a gun", {"M9", "Remington 870", "AK-47"}, function(v)
        local A_1 = game:GetService("Workspace")["Prison_ITEMS"].giver[v].ITEMPICKUP
        local Event = game:GetService("Workspace").Remote.ItemHandler
        Event:InvokeServer(A_1)
    end)
    
    MainS:NewDropdown("Gun Mod", "makes the gun op", {"M9", "Remington 870", "AK-47"}, function(v)
        local module = nil
        if game:GetService("Players").LocalPlayer.Backpack:FindFirstChild(v) then
            module = require(game:GetService("Players").LocalPlayer.Backpack[v].GunStates)
        elseif game:GetService("Players").LocalPlayer.Character:FindFirstChild(v) then
            module = require(game:GetService("Players").LocalPlayer.Character[v].GunStates)   
        end
        if module ~= nil then
            module["MaxAmmo"] = math.huge
            module["CurrentAmmo"] = math.huge
            module["StoredAmmo"] = math.huge
            module["FireRate"] = 0.000001
            module["Spread"] = 0
            module["Range"] = math.huge
            module["Bullets"] = 10
            module["ReloadTime"] = 0.000001
            module["AutoFire"] = true
        end
    end)
    local Player = Window:NewTab("Player")
    local PlayerS = Player:NewSection("Player")

    PlayerS:NewSlider("WalkSpeed", "Changes The Speed", 250, 0, function(v) -- 500 (MaxValue) | 0 (MinValue)
        game.Players.LocalPlayer.Character.Humanoid.WalkSpeed = v
    end)
    PlayerS:NewSlider("JumpPower", "Changes The JumpPower", 250, 0, function(v) -- 500 (MaxValue) | 0 (MinValue)
        game.Players.LocalPlayer.Character.Humanoid.JumpPower = v
    end)

    	local Setting = Window:NewTab("Settings")
      local SettingS = Setting:NewSection("Settings")

      SettingS:NewKeybind("KeyBind", "To Hide", Enum.KeyCode.P, function()
          Library:ToggleUI()
      end)

print("Check Point 1 ")
      
      

end







if game.PlaceId == 13822889 then

    local Library = loadstring(game:HttpGet("https://raw.githubusercontent.com/xHeptc/Kavo-UI-Library/main/source.lua"))()
    local Window = Library.CreateLib("Lumber Tycoon 2", "Midnight")
    

    
    local Player = Window:NewTab("Player")
    local PlayerS = Player:NewSection("Player")

    PlayerS:NewSlider("WalkSpeed", "Changes The Speed", 250, 0, function(v) -- 500 (MaxValue) | 0 (MinValue)
        game.Players.LocalPlayer.Character.Humanoid.WalkSpeed = v
    end)
    PlayerS:NewSlider("JumpPower", "Changes The JumpPower", 250, 0, function(v) -- 500 (MaxValue) | 0 (MinValue)
        game.Players.LocalPlayer.Character.Humanoid.JumpPower = v
    end)



    PlayerS:NewToggle("Water Soild", "True or false State ment", function(state)
        if state then
            local value = true
            game:GetService("Workpace").Bridge.VerticalLiftBridge.WaterModel.Water.CanCollide = value
            game:GetService("Workspace").Water.CanCollide = value
        else
           local value = false
           game:GetService("Workpace").Bridge.VerticalLiftBridge.WaterModel.Water.CanCollide = value
           game:GetService("Workspace").Water.CanCollide = value
        end
    end)
    PlayerS:NewButton("All Blue Prints", "Gives all bluepritns", function()
        for _,v in pairs(game:GetService("ReplicatedStorage").Purchasables.Structures.BlueprintStructures:GetChildren()) do
            if game:GetService("Players").LocalPlayer.PlayerBlueprints.Blueprints:FindFirstChild(v.Name) == nil then
                local folder = Instance.new("Folder", game:GetService("Players").LocalPlayer.PlayerBlueprints.Blueprints)
                folder.Name = v.Name
                local category = Instance.new("StringValue", folder)
                category.Name = "ItemCategory"
                category.Value = v.ItemCategory.Value
            end
        end
    end)









    local Dupe = Window:NewTab("Dupes")
    local Dupes = Dupe:NewSection("Dupes")



    	local Setting = Window:NewTab("Settings")
      local SettingS = Setting:NewSection("Settings")

      SettingS:NewKeybind("KeyBind", "To Hide", Enum.KeyCode.P, function()
        Library:ToggleUI()
      end)  






      print("Check Point 2")
end



function NotSupported()
      local Library = loadstring(game:HttpGet("https://raw.githubusercontent.com/xHeptc/Kavo-UI-Library/main/source.lua"))()
    local Window = Library.CreateLib("Unknow Game", "Midnight")
    

    
    local Player = Window:NewTab("Player")
    local PlayerS = Player:NewSection("Player")

    PlayerS:NewSlider("WalkSpeed", "Changes The Speed", 250, 0, function(v) -- 500 (MaxValue) | 0 (MinValue)
        game.Players.LocalPlayer.Character.Humanoid.WalkSpeed = v
    end)
    PlayerS:NewSlider("JumpPower", "Changes The JumpPower", 250, 0, function(v) -- 500 (MaxValue) | 0 (MinValue)
        game.Players.LocalPlayer.Character.Humanoid.JumpPower = v
    end)



    	local Setting = Window:NewTab("Settings")
      local SettingS = Setting:NewSection("Settings")

      SettingS:NewKeybind("KeyBind", "To Hide", Enum.KeyCode.P, function()
        Library:ToggleUI()
      end)  
    
    SettingS:NewTextBox(""

    
    game.StarterGui:SetCore("SendNotification", {
  Title = "Banana Hub";
  Text = "Game is not supported";
  Button1 = "";
  Duration = "4";
})
end



local Creater = "Banana-nana on github"


NotSupported()

game.StarterGui:SetCore("SendNotification", {
  Title = "Banana Hub Beta 9";
  Text = "thanks 4 using it";
  Button1 = "Done";
  Duration = "4";
})

print("Last CheckPoint")
